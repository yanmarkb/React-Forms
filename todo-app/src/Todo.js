import React from "react";

const Todo = ({ id, task, removeTodo }) => {
	const handleRemove = () => removeTodo(id);

	return (
		<div>
			<div>{task}</div>
			<button onClick={handleRemove}>X</button>
		</div>
	);
};

export default Todo;
