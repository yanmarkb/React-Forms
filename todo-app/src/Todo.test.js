import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

test("Todo component renders without crashing", () => {
	render(<Todo id="1" task="Test Task" removeTodo={() => {}} />);
});

test("Todo component matches snapshot", () => {
	const { asFragment } = render(
		<Todo id="1" task="Test Task" removeTodo={() => {}} />
	);
	expect(asFragment()).toMatchSnapshot();
});

test("Todo component calls removeTodo function when button is clicked", () => {
	const removeTodo = jest.fn();
	const { getByText } = render(
		<Todo id="1" task="Test Task" removeTodo={removeTodo} />
	);
	const button = getByText("X");
	fireEvent.click(button);
	expect(removeTodo).toHaveBeenCalled();
});
