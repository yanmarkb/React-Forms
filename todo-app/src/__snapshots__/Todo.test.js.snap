// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Test Task
    </div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
